### PR TITLE
[dart] Support/Fix sourceFolder parameter and docs

### LINKED ISSUE #10646: [dart-dio-next] How to have more control over the sourceFolder?
#### Description

There's a config option sourceFolder metnioned in the config options for dart-dio-next: https://openapi-generator.tech/docs/generators/dart-dio-next/

However, when I set the folder in my yaml config file, it always appends lib/src to it. E.g. I want my client source code to end up in the /lib/client package. If I put this in the sourceFolder, the result I end up with is /lib/client/lib/src/. Using lib/client instead doesn't help.
Tried putting just client in my config yaml file, and then adding -o ./lib to my command line but that doesn't help either. 

Looking at the code here: https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioNextClientCodegen.java
On line 184 and further, I assume those lib and src strings are being hardcoded in the sourceFolder, correct?

Can this be changed, so we actually have the option to set the sourceFolder as we want it?

##### openapi-generator version

Latest one: openapi-generator-cli-5.3.0-20210907.065732-24.jar

##### OpenAPI declaration file content or url

The config yaml file I use:

##### Config options for the dart-dio-next generator
pubName: MyProject
pubVersion: 0.0.1
sourceFolder: lib/client

##### Command line used for generation

java -DapiTests=false -DmodelTests=false -DapiDocs=false -DmodelDocs=false -jar openapi-generator-cli.jar generate -i openapi.yaml -g dart-dio-next -c open-generator-config.yaml

#### Steps to reproduce
* Add a sourceFolder to the config yaml file
* Run the generator
* See that the generated files will always be in <sourceFolder>/lib/src

### PR DESCRIPTION
**Pull request by kuhnroyal**
* allow passing the sourceFolder to generators
* not sure how this translates to the dart2 generator due to the `part` files but defaults work like before
* fix doc generation not showing default value of CLI options
* fixes #10646

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

CC @jaumard (2018/09) @josh-burton (2019/12) @amondnet (2019/12) @sbu-WBT (2020/12) @kuhnroyal (2020/12) @agilob (2020/12) @ahmednfwela (2021/08)

### CODE REVIEW

**Review by josh-burton — APPROVED** [review]

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java [hunk 1]
```diff
@@ -271,8 +279,10 @@ public void processOpts() {
         }
 
         if (additionalProperties.containsKey(CodegenConstants.SOURCE_FOLDER)) {
-            this.setSourceFolder((String) additionalProperties.get(CodegenConstants.SOURCE_FOLDER));
+            String srcFolder = (String) additionalProperties.get(CodegenConstants.SOURCE_FOLDER);
+            this.setSourceFolder(srcFolder.replace('/', File.separatorChar));
         }
+        additionalProperties.put(CodegenConstants.SOURCE_FOLDER, sourceFolder);
```
Comment (agilob) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java:285]: is this null safe?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java [hunk 2]
```diff
@@ -62,6 +61,7 @@ public DartDioClientCodegen() {
 
         outputFolder = "generated-code/dart-dio";
         embeddedTemplateDir = "dart-dio";
+        sourceFolder = "";
```
Comment (agilob) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java:64]: can this be deduplicated into abstractdart?

**Review by agilob — APPROVED** [review]

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java (hunk 2 above)
Comment (kuhnroyal) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java:64]: This client is being deleted on the other branch :)

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java (hunk 1 above)
Comment (kuhnroyal) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java:285]: I think so, I have not found a way to pass `null` via a property (file/CLI) - it is always an empty String

**Review by agilob — APPROVED** [review]

**Review by agilob — APPROVED** [review]

**Pull request merged**